src/main.ts

```diff
@@ -22,8 +24,24 @@ export const run = async () => {
         }
 
         let inlineScript: string = core.getInput('inlineScript', { required: true });
-        let azcliversion: string = core.getInput('azcliversion', { required: true }).trim().toLowerCase();
-       
+        let azcliversion: string = core.getInput('azcliversion', { required: false }).trim().toLowerCase();
+
+        if(azcliversion == AZ_CLI_VERSION_DEFAULT_VALUE){
+            const { stdout, stderr } = await cpExec('az version');
+            if (!stderr) {

```

Comment:
`stderr` should never be used as an indicator of failure. See https://github.com/Azure/azure-cli/issues/18372.